assets/js/common/forms.js

```diff
@@ -239,6 +239,48 @@ function showLocalSaveWarning() {
     }
 }
 
+function animateSaveButtons() {
+    $('.js-save-btn-form').on('submit', function(event) {
+        event.preventDefault();
+
+        const $form = $(this);
+        const $btn = $form.find('.js-save-btn');
+        const $label = $btn.find('.js-save-btn-label');
+
+        // animation is 0.3s long x 3 (with a 0.3s delay, eg. 1.2 total)
+        const cssDotAnimationDuration = 300 * 4;
+        const cssIconAnimationDuration = 300;
+
+        const text = {
+            interstitial: $btn.data('interstitial'),
+            complete: $btn.data('complete')
+        };
+
+        const classes = {
+            loading: 'is-loading',
+            complete: 'is-complete'
+        };
+
+        const setBtnLabel = text => {
+            $label.text(text);
+        };
+
+        $btn.addClass(classes.loading);
+        setBtnLabel(text.interstitial);
+
+        window.setTimeout(function() {
+            setBtnLabel(text.complete);
+            $btn.addClass(classes.complete);
+        }, cssDotAnimationDuration);
+
+        window.setTimeout(function() {
+            // Remove this event binding and re-trigger the submit request
+            $form.off(event);
+            $form.submit();
+        }, cssDotAnimationDuration + cssIconAnimationDuration);
+    });
+}
+
 function init() {
```

Comment:
General comment, at some point we should probably restructure things so that the Vue form code and "non-Vue" form code lives closer together, not a problem for now.